### PR TITLE
Add HTMLBaseElement.href docs

### PR DESCRIPTION
**Pull request by teoli2003**
This adds the missing `HTMLBaseElement.href` page.

Part of the project to document all interoperable features ([mdn/discussions#476](https://github.com/orgs/mdn/discussions/476))

### CODE REVIEW

File: files/en-us/web/api/htmlbaseelement/href/index.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+title: "HTMLBaseElement: href property"
+short-title: href
+slug: Web/API/HTMLBaseElement/href
+page-type: web-api-instance-property
+browser-compat: api.HTMLBaseElement.href
+---
+
+{{APIRef("HTML DOM")}}
+
+The **`HTMLBaseElement.href`** property of the {{domxref("HTMLBaseElement")}} interface contains a string that is the URL to use as the base for relative URLs, like links pointing to a fragment in the document (`<a href="#example">``).
```
Comment (skyclouds2001) [inline files/en-us/web/api/htmlbaseelement/href/index.md:11]:
```suggestion
The **`href`** property of the {{domxref("HTMLBaseElement")}} interface contains a string that is the URL to use as the base for relative URLs, like links pointing to a fragment in the document (`<a href="#example">``).
```

also https://github.com/mdn/content/pull/31462#discussion_r1440445929

File: files/en-us/web/api/htmlbaseelement/href/index.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+title: "HTMLBaseElement: href property"
+short-title: href
+slug: Web/API/HTMLBaseElement/href
+page-type: web-api-instance-property
+browser-compat: api.HTMLBaseElement.href
+---
+
+{{APIRef("HTML DOM")}}
+
+The **`href`** property of the {{domxref("HTMLBaseElement")}} interface contains a string that is the URL to use as the base for relative URLs, like links pointing to a fragment in the document (`<a href="#example">``).
```
Comment (teoli2003) [inline files/en-us/web/api/htmlbaseelement/href/index.md:11]:
```suggestion
The **`href`** property of the {{domxref("HTMLBaseElement")}} interface contains a string that is the URL to use as the base for relative URLs, like links pointing to a fragment in the document (`<a href="#example">`).
```

File: files/en-us/web/api/htmlbaseelement/href/index.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+A string that contains a URL, or the empty string (`""`) if there is no `href` element.
+
+## Examples
+
+```html
+<base href="https://developer.mozilla.org/example" />
+```
+
+```js
+const base = document.getElementsByTag("base")[0];
```
Comment (wbamberg) [inline files/en-us/web/api/htmlbaseelement/href/index.md:26]:
```suggestion
const base = document.getElementsByTagName("base")[0];
```

File: files/en-us/web/api/htmlbaseelement/href/index.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+{{APIRef("HTML DOM")}}
+
+The **`href`** property of the {{domxref("HTMLBaseElement")}} interface contains a string that is the URL to use as the base for relative URLs, like links pointing to a fragment in the document (`<a href="#example">`).
+
+It reflects the `href` attribute of the {{HTMLElement("base")}} element and is the empty string (`""`) if there are none.
+
+## Value
+
+A string that contains a URL, or the empty string (`""`) if there is no `href` element.
```
Comment (wbamberg) [inline files/en-us/web/api/htmlbaseelement/href/index.md:17]:
```suggestion
A string that contains a URL, or the empty string (`""`) if the `<base>` element does not include the `href` attribute.
```

? Although I don't get this. For a page with a `<base>` element like `<base target="_blank" />`, served from "http://[::]:8000/", I get:

```
const base = document.getElementsByTagName("base")[0];
console.log(`base href:  ${base.href}`);
> base href:	http://[::]:8000/
```

File: files/en-us/web/api/htmlbaseelement/href/index.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+## Examples
+
+```html
+<base href="https://developer.mozilla.org/example" />
+```
+
+```js
+const base = document.getElementsByTag("base")[0];
+console.log(base.href); // 'https://developer.mozilla.org/example'
+```
```
Comment (wbamberg) [inline files/en-us/web/api/htmlbaseelement/href/index.md:28]: Should we also show the result if there is no `href` attribute?

**Review by wbamberg — DISMISSED** [review]

File: files/en-us/web/api/htmlbaseelement/href/index.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+title: "HTMLBaseElement: href property"
+short-title: href
+slug: Web/API/HTMLBaseElement/href
+page-type: web-api-instance-property
+browser-compat: api.HTMLBaseElement.href
+---
+
+{{APIRef("HTML DOM")}}
+
+The **`href`** property of the {{domxref("HTMLBaseElement")}} interface contains a string that is the URL to use as the base for relative URLs, like links pointing to a fragment in the document (`<a href="#example">`).
```
Comment (estelle) [inline files/en-us/web/api/htmlbaseelement/href/index.md:11]:
```suggestion
The **`href`** property of the {{domxref("HTMLBaseElement")}} interface contains a string that is the URL to use as the base for relative URLs, such as links pointing to a fragment in the document (`<a href="#example">`).
```

File: files/en-us/web/api/htmlbaseelement/href/index.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+It reflects the `href` attribute of the {{HTMLElement("base")}} element and is the empty string (`""`) if there are none.
```
Comment (estelle) [inline files/en-us/web/api/htmlbaseelement/href/index.md:13]:
```suggestion
It reflects the `href` attribute of the {{HTMLElement("base")}} element.
```

File: files/en-us/web/api/htmlbaseelement/href/index.md (hunk 5 above)
Comment (estelle) [reply files/en-us/web/api/htmlbaseelement/href/index.md:28]: there can only be one base element in a page, so we can't include two in a single example; opting to not include and `target`-only `<base>` example.

File: files/en-us/web/api/htmlbaseelement/href/index.md (hunk 6 above)
Comment (estelle) [inline files/en-us/web/api/htmlbaseelement/href/index.md:11]:
```suggestion
The **`href`** property of the {{domxref("HTMLBaseElement")}} interface contains a string that is the URL to use as the base for [relative URLs](/en-US/docs/Learn/Common_questions/Web_mechanics/What_is_a_URL#absolute_urls_vs._relative_urls), such as links pointing to a fragment in the document (`<a href="#example">`).
```

**Review by estelle — COMMENTED** [review]
edits per review

**Comment by estelle** [comment]
@wbamberg when you have a chance, a re-review would be great.

**Review by estelle — APPROVED** [review]

File: files/en-us/web/api/htmlbaseelement/href/index.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+{{APIRef("HTML DOM")}}
+
+The **`href`** property of the {{domxref("HTMLBaseElement")}} interface contains a string that is the URL to use as the base for [relative URLs](/en-US/docs/Learn/Common_questions/Web_mechanics/What_is_a_URL#absolute_urls_vs._relative_urls), such as links pointing to a fragment in the document (`<a href="#example">`).
+
+It reflects the `href` attribute of the {{HTMLElement("base")}} element.
+
+## Value
+
+A string that contains a URL, or the empty string (`""`) if the `<base>` element does not include the `href` attribute.
```
Comment (hamishwillee) [inline files/en-us/web/api/htmlbaseelement/href/index.md:17]:
```suggestion
A string that contains a URL, or the empty string (`""`) if the corresponding `<base>` element does not include the `href` attribute.
```

File: files/en-us/web/api/htmlbaseelement/href/index.md (hunk 5 above)
Comment (hamishwillee) [reply files/en-us/web/api/htmlbaseelement/href/index.md:53]: We could have two examples. I'd have done that, except that this violates CSP and I get an error for that case.

File: files/en-us/web/api/htmlbaseelement/href/index.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+title: "HTMLBaseElement: href property"
+short-title: href
+slug: Web/API/HTMLBaseElement/href
+page-type: web-api-instance-property
+browser-compat: api.HTMLBaseElement.href
+---
+
+{{APIRef("HTML DOM")}}
+
+The **`href`** property of the {{domxref("HTMLBaseElement")}} interface contains a string that is the URL to use as the base for [relative URLs](/en-US/docs/Learn/Common_questions/Web_mechanics/What_is_a_URL#absolute_urls_vs._relative_urls), such as links pointing to a fragment in the document (`<a href="#example">`).
```
Comment (hamishwillee) [inline files/en-us/web/api/htmlbaseelement/href/index.md:11]: I see this as unnecessary, as indicated by Will, in particular now this links to the relative URLs page.
```suggestion
The **`href`** property of the {{domxref("HTMLBaseElement")}} interface contains a string that is the URL to use as the base for [relative URLs](/en-US/docs/Learn/Common_questions/Web_mechanics/What_is_a_URL#absolute_urls_vs._relative_urls).
```

**Review by hamishwillee — APPROVED** [review]
I've made the example into a live example. Merging as this is now IMO sufficient (and Will is on holiday).

File: files/en-us/web/api/htmlbaseelement/href/index.md (hunk 5 above)
Comment (hamishwillee) [reply files/en-us/web/api/htmlbaseelement/href/index.md:53]: But I think is OK without both. The expected behaviour is stated.

**Pull request merged**